### ISSUE TITLE
Events not caught during tests after evm_revert on ganache

### ISSUE THREAD

**Issue by Silur** [trufflesuite/truffle#5149]
EVM events are not caught after restoring a saved snapshot of the ganache state during tests with truffle:

```javascript
const { snapshot } = require('@openzeppelin/test-helpers');
const MyContract = artifacts.require('MyContract');
contract("MyContract", function (accounts) {
  before(async () => {
    this.myContract = await MyContract.deployed();
    this.snapA = await snapshot();
  })
  it("should work", async () => {
   await myContract.throwEvent(); // <-- event is shown during `truffle test --show-events`
  })
  it("should work too", async () => {
   await this.snapA.restore();
   await myContract.throwEvent(); // <-- event is NOT shown during `truffle test --show-events`
  })
});
```
However, checking the transaction receipt manually has the logs set up with the event data in question.

Tested with:

Truffle v5.5.0 (core: 5.5.0)
Ganache v7.0.1
Solidity - 0.8.11 (solc-js)
Node v16.14.0
Web3.js v1.5.3

**Comment by davidmurdoch** [comment]
Thanks for opening this, @silur. Can you try the latest version of truffle and let us know if that fixes either of the issues (the events or the revert issue)?

**Comment by cliffoo** [comment]
Closing for Issue Maintenance.
